rust/macros/pin_data.rs

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use proc_macro::{Punct, Spacing, TokenStream, TokenTree};
+
+pub(crate) fn pin_data(args: TokenStream, input: TokenStream) -> TokenStream {
+    // This proc-macro only does some pre-parsing and then delegates the actual parsing to
+    // `kernel::_pin_data!`.
```

Comment:
There is a typo here, should be `__pin_data`, right?

Reply:
That is correct, good catch!